server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/heuristic/ChiSquareTests.java

```diff
@@ -8,21 +8,17 @@
 
 package org.elasticsearch.search.aggregations.bucket.terms.heuristic;
 
-import org.elasticsearch.search.aggregations.bucket.terms.AbstractSignificanceHeuristicTests;
+import org.elasticsearch.search.aggregations.bucket.AbstractNXYSignificanceHeuristicTestCase;
 
-public class ChiSquareTests extends AbstractSignificanceHeuristicTests {
-    @Override
-    protected SignificanceHeuristic getHeuristic() {
-        return new ChiSquare(randomBoolean(), randomBoolean());
-    }
+public class ChiSquareTests extends AbstractNXYSignificanceHeuristicTestCase {
 
     @Override
-    protected boolean testZeroScore() {
-        return false;
+    public void testAssertions() {
+        testBackgroundAssertions(new ChiSquare(true, true), new ChiSquare(true, false));
     }
 
     @Override
-    public void testAssertions() {
-        testBackgroundAssertions(new ChiSquare(true, true), new ChiSquare(true, false));
+    protected SignificanceHeuristic getHeuristic(boolean includeNegatives, boolean backgroundIsSuperset) {
+        return new ChiSquare(includeNegatives, backgroundIsSuperset);
     }
```

Comment:
...okay github, that's an interesting way to interpret that change

Reply:
yeah, sorry :(. It looked cleaner on a local `git diff`

Reply:
Not your fault.  I think github optimizes for minimal edit distance, not fewest lines changed.  I got the point though, and that's all that matters.